bootloader/src/main.rs

```diff
@@ -373,6 +373,7 @@ fn load_kernel(
             MEMORY_MAP_ALLOCATED_SIZE,
         )
     };
+    memory_map_storage.fill(0);
```

Comment:
amazing find!

Reply:
one question: do you think you would need to do this kind of thing again? I.e. allocate some pages and fill with zero?

Reply:
I'm pretty sure we don't do it at th moment except in the page tables, but I'm pretty sure they are initialized slightly differently. Maybe if we add a few more usages it would be worth it but I would like to merge as-is if you don't mind.